xlml/apis/task.py

```diff
@@ -77,6 +77,39 @@ def run(self) -> DAGNode:
 
     return group
 
+  def run_with_run_name_generation(self) -> DAGNode:
+    """Run a test job.
+
+    Returns:
+      A task group with the following tasks chained: provision, run_model,
+      post_process and clean_up.
+    """
+    with TaskGroup(
+        group_id=self.task_test_config.benchmark_id, prefix_group_id=True
+    ) as group:
+      run_name = maxtext.generate_run_name(self.task_test_config.benchmark_id)
+      tb_file_location = maxtext.generate_tb_file_location(
+          run_name, self.task_metric_config.tensorboard_summary.file_location
+      )
+
+      # Set run_name in run_model_cmds
+      new_run_model_cmds = [f"export M_RUN_NAME={run_name}"]
+      for cmd in self.task_test_config.run_model_cmds:
+        new_run_model_cmds.append(cmd)
+      self.task_test_config.run_model_cmds = new_run_model_cmds
```

Comment:
Ah, this part does seem MaxText-specific... Is there any way around modifying the run_model_cmds to communicate the run name to the workload? cc @RissyRan for her AirFlow expertise, maybe set a variable here and consume it in the workload?

Reply:
Yeah I agree this part is MaxText specific. I think the only way I could think of to keep the generality here was to have the `prepend_tasks` where we could contain the MaxText specific tasks/operations and then prepend it to the main job. I would definitely be open to other ideas or workarounds.

Reply:
Yeah, I think I get your point Jon. You may want to add a global variable [here](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/xlml/apis/test_config.py#L184) to add this export directly. Currently, I did not see other use cases to make changes to run_model_cmds besides maxtext runs. As we already have a `run_with_run_name_generation` helper here to handle the case. I am fine with current implementation :)

Reply:
Sounds good. I am fine with the current implementation as well, and we can modify it later if issues arise.